src/client.ts

```diff
@@ -1,10 +1,33 @@
 import { PineconeConfigurationError } from './errors';
-import type { ClientConfiguration } from './pinecone';
+import type { ClientConfiguration } from './types';
+import type { ConfigurationParameters as IndexOperationsApiConfigurationParameters } from './pinecone-generated-ts-fetch';
+import {
+  IndexOperationsApi,
+  Configuration as ApiConfiguration,
+} from './pinecone-generated-ts-fetch';
+import { describeIndex, listIndexes } from './control';
 
 export class Client {
   private config: ClientConfiguration;
 
   constructor(options: ClientConfiguration) {
+    this._validateConfig(options);
+
+    this.config = options;
+
+    const { apiKey, environment } = options;
+    const controllerPath = `https://controller.${environment}.pinecone.io`;
+    const apiConfig: IndexOperationsApiConfigurationParameters = {
+      basePath: controllerPath,
+      apiKey,
+    };
+    const api = new IndexOperationsApi(new ApiConfiguration(apiConfig));
+
+    this['describeIndex'] = describeIndex(api);
+    this['listIndexes'] = listIndexes(api);
```

Comment:
I initially used this `this['describeIndex']` syntax without thinking about it too much mainly because my in-editor typescript checking was getting mad about me writing `this.describeIndex = describeIndex(api)` with an error like `Property 'describeIndex' does not exist on type 'Client'`. But I later learned that by declaring these properties like this I could satisfy the typescript checker:

```
export class Client {
    describeIndex: ReturnType<typeof describeIndex>;
    listIndexes: ReturnType<typeof listIndexes>;

    constructor(options: ClientConfiguration) {
        // ... other code
        this.describeIndex = describeIndex(api);
        this.listIndexes = listIndexes(api);
    }
```

I totally agree these things all need to be documented. I have a separate task/mini-project/OKR about docs automation. My current plan is to try out jsdoc. If above format proves difficult with this tool then I'll revisit.